src/contracts/LiquidationEngine.sol

```diff
@@ -175,10 +180,9 @@ contract LiquidationEngine is
 
         if (_collateralToSell == 0) revert LiqEng_NullCollateralToSell();
 
-        if (
-          _limitAdjustedDebt != _safeData.generatedDebt
-            && (_safeData.generatedDebt - _limitAdjustedDebt) * _safeEngCData.accumulatedRate < _debtFloor
-        ) revert LiqEng_DustySAFE();
+        if (currentOnAuctionSystemCoins + _limitAdjustedDebt > _params.onAuctionSystemCoinLimit) {
+          revert LiqEng_ExceededCapacity();
+        }
```

Comment:
this is the only line where we should consider the `onAuctionSystemCoinLimit`, please remove it from the 1st `limitAdjustedDebt` calculation.
if the limit is reached, we don't cap the amount to liquidate, we just revert 🙏